### PR TITLE
docs(microservices): add gRPC Metadata examples

### PR DESCRIPTION
**Pull request by AlexDaSoul**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/docs.nestjs.com/blob/master/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: https://github.com/nestjs/nest/pull/5143#issuecomment-665585769

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: content/microservices/grpc.md [hunk 1]
```diff
@@ -158,7 +160,7 @@ Both `@GrpcMethod()` decorator arguments are optional. If called without the sec
 @Controller()
 export class HeroesController {
   @GrpcMethod('HeroesService')
-  findOne(data: HeroById, metadata: any): Hero {
+  findOne(data: HeroById, metadata: any, call: any): Hero {
```
Comment (kamilmysliwiec) [inline content/microservices/grpc.md:163]: Can we use types for `metadata` and `call`? I believe that gRPC package exposes some type definitions we could import here

Comment (AlexDaSoul) [reply content/microservices/grpc.md:163]: We can use Metadata from grpc.Metadata, but idk what type for call.

Comment (kamilmysliwiec) [reply content/microservices/grpc.md:163]: It depends on the type of handler. 

For a unary call (by default), it should be `grpc.ServerUnaryCall<RequestType>`.
For client streaming - `grpc.ServerReadableStream<RequestType>`
For server streaming - `grpc.ServerWritableStream<RequestType>`
For bidi streaming (duplex) - `grpc.ServerDuplexStream<RequestType, ResponseType>`

We could just use `any` for `RequestType`/`ResponseType` type arguments to slightly simplify this.

Comment (AlexDaSoul) [reply content/microservices/grpc.md:163]: This is handleCall<RequestType, ResponseType>

`  type handleCall<RequestType, ResponseType> =
    handleUnaryCall<RequestType, ResponseType> |
    handleClientStreamingCall<RequestType, ResponseType> |
    handleServerStreamingCall<RequestType, ResponseType> |
    handleBidiStreamingCall<RequestType, ResponseType>;`

But I doubt is needed for this doc cause is hard. If you think that it needs to be added then I will

Comment (AlexDaSoul) [reply content/microservices/grpc.md]: fixed

File: content/microservices/grpc.md [hunk 2]
```diff
@@ -440,3 +442,68 @@ lotsOfGreetings(requestStream: any, callback: (err: unknown, value: HelloRespons
...
+      { id: 2, name: 'Doe' },
+    ];
+
+    srvMetadata.add('Set-Cookie', 'yummy_cookie=choco');
+    call.sendMetadata(srvMetadata);
+
+    return items.find(({ id }) => id === data.id);
+  }
+}
+```
```
Comment (jmcdo29) [inline content/microservices/grpc.md:489]: This looks great. Can we add a hint on where to `import` `Metadata` from [similar to here](https://docs.nestjs.com/graphql/resolvers#code-first-resolver) (line 157 of graphql/resovler-map, GitHub won't link the markdown file properly)

**Review by jmcdo29 — CHANGES_REQUESTED** [review]

File: content/microservices/grpc.md (hunk 2 above)
Comment (AlexDaSoul) [reply content/microservices/grpc.md:489]: done

**Pull request merged**